### PR TITLE
RVizVisualizer update

### PR DESCRIPTION
**Pull request by EtienneAr**
Changes are : 
* Give the option to the user to not init a ros node in **initViewer** (if one is already started) to prevent a crash
* Re-implement **clean** method to actually follow spec
* So small bug fix

### CODE REVIEW

**Review by jcarpent — APPROVED** [review]
Looks good to me.

**Comment by jcarpent** [comment]
Thanks a lot @EtienneAr for fixing these bugs

File: bindings/python/pinocchio/visualize/rviz_visualizer.py [hunk 1]
```diff
@@ -70,7 +70,8 @@ def initViewer(self, viewer=None, windowName="python-pinocchio", loadModel=False
                           category=UserWarning, stacklevel=2)
             return None
 
-        init_node('pinocchio_viewer', anonymous=True, log_level=WARN)
+        if(initRosNode):
```
Comment (nim65s) [inline bindings/python/pinocchio/visualize/rviz_visualizer.py:73]:
```suggestion
        if initRosNode:
```

File: bindings/python/pinocchio/visualize/rviz_visualizer.py [hunk 2]
```diff
@@ -116,26 +117,31 @@ def loadViewerModel(self, rootNodeName="pinocchio"):
         self.display()
 
     def clean(self):
-        """ Delete all the objects from the whole scene """
-        self.viewer.app.quit()
-        self.viewer.app = None
-        self.viewer.viz = None
-        self.viewer.viz_manager = None
+        """Delete all the objects from the whole scene """
+        if(hasattr(self, 'collisions_publisher')):
```
Comment (nim65s) [inline bindings/python/pinocchio/visualize/rviz_visualizer.py:121]:
```suggestion
        if hasattr(self, 'collisions_publisher'):
```

File: bindings/python/pinocchio/visualize/rviz_visualizer.py [hunk 3]
```diff
@@ -116,26 +117,31 @@ def loadViewerModel(self, rootNodeName="pinocchio"):
...
+            self._clean(self.collisions_publisher)
+            self.collision_ids = []
+
+        if(hasattr(self, 'visuals_publisher')):
```
Comment (nim65s) [inline bindings/python/pinocchio/visualize/rviz_visualizer.py:125]:
```suggestion
        if hasattr(self, 'visuals_publisher'):
```

File: bindings/python/pinocchio/visualize/rviz_visualizer.py [hunk 4]
```diff
@@ -116,26 +117,31 @@ def loadViewerModel(self, rootNodeName="pinocchio"):
...
 
     def display(self, q = None):
         """Display the robot at configuration q in the viz by placing all the bodies."""
         # Update the robot kinematics and geometry.
         if q is not None:
             pin.forwardKinematics(self.model,self.data,q)
 
-        pin.updateGeometryPlacements(self.model, self.data, self.collision_model, self.collision_data)
-        self.collision_ids = self.plot(self.collisions_publisher, self.collision_model, self.collision_data, self.collision_ids)
+        if self.collision_model != None and hasattr(self, 'collisions_publisher'):
```
Comment (nim65s) [inline bindings/python/pinocchio/visualize/rviz_visualizer.py:135]:
```suggestion
        if self.collision_model is not None and hasattr(self, 'collisions_publisher'):
```

File: bindings/python/pinocchio/visualize/rviz_visualizer.py [hunk 5]
```diff
@@ -116,26 +117,31 @@ def loadViewerModel(self, rootNodeName="pinocchio"):
...
+            pin.updateGeometryPlacements(self.model, self.data, self.collision_model, self.collision_data)
+            self.collision_ids = self._plot(self.collisions_publisher, self.collision_model, self.collision_data, self.collision_ids)
 
-        pin.updateGeometryPlacements(self.model, self.data, self.visual_model, self.visual_data)
-        self.visual_ids = self.plot(self.visuals_publisher, self.visual_model, self.visual_data, self.visual_ids)
+        if self.visual_model != None and hasattr(self, 'visuals_publisher'):
```
Comment (nim65s) [inline bindings/python/pinocchio/visualize/rviz_visualizer.py:139]:
```suggestion
        if self.visual_model is not None and hasattr(self, 'visuals_publisher'):
```

**Review by nim65s — COMMENTED** [review]
Just some details :)

**Comment by EtienneAr** [comment]
Thank you @nim65s

**Pull request merged**